src/Query/Filter/DateField.ts

```diff
@@ -199,4 +200,20 @@ export abstract class DateField extends Field {
             return compareByDate(this.date(a), this.date(b));
         };
     }
+
+    public supportsGrouping(): boolean {
+        return true;
+    }
+
+    protected static readonly groupDateFormat = 'YYYY-MM-DD dddd';
```

Comment:
@claremacrae not sure if declaring a `protected static readonly` here is ok (the other option is on top of the class), please advise.

Reply:
This comment made me look more closely at `HappensDateField`:

https://github.com/obsidian-tasks-group/obsidian-tasks/blob/04b0303f95f44eb4e3ec14503f6e68b15dadd66c/src/Query/Filter/HappensDateField.ts#L29-L35

Which means that the `DateField` implementation of grouping _would_ work fine for HappensDateField...

Following that through, removing the grouping code from `HappensDateField` means that `groupDateFormat` can be inlined again, and you don't have to worry about where/how to declare it!

Hope that makes some kind of sense.

Reply:
oh right. thanks!